runtime/protocol/aws-json-protocols/common/src/aws/smithy/kotlin/runtime/awsprotocol/json/RestJsonErrorDeserializer.kt

```diff
@@ -46,15 +46,17 @@ public object RestJsonErrorDeserializer {
     }
 
     public fun deserialize(headers: Headers, payload: ByteArray?): ErrorDetails {
-        var code: String? = headers[X_AMZN_ERROR_TYPE_HEADER_NAME]
-        var message: String? = if (headers[X_AMZN_ERROR_MESSAGE_HEADER_NAME] == null) headers[X_AMZN_EVENT_ERROR_MESSAGE_HEADER_NAME] else headers[X_AMZN_ERROR_MESSAGE_HEADER_NAME]
+        var message = headers[X_AMZN_ERROR_MESSAGE_HEADER_NAME] ?: headers[X_AMZN_ERROR_MESSAGE_HEADER_NAME]
```

Comment:
Should this first header parsed be `X_AMZN_EVENT_ERROR_MESSAGE_HEADER_NAME` instead? Right now they're both `X_AMZN_ERROR_MESSAGE_HEADER_NAME`